src/lib/operators/table_scan/column_is_null_table_scan_impl.hpp

```diff
@@ -17,39 +17,49 @@ class BaseValueSegment;
 // Scans for the presence or absence of NULL values in a given column. This is not a
 // AbstractDereferencedColumnTableScanImpl because that super class drops NULL values in the referencing column, which
 // would break the `NOT NULL` scan.
-class ColumnIsNullTableScanImpl : public AbstractTableScanImpl {
+class ColumnIsNullTableScanImpl : public AbstractDereferencedColumnTableScanImpl {
  public:
   ColumnIsNullTableScanImpl(const std::shared_ptr<const Table>& in_table, const ColumnID column_id,
-                            const PredicateCondition& predicate_condition);
+                            const PredicateCondition& init_predicate_condition);
 
-  std::string description() const override;
-
-  std::shared_ptr<RowIDPosList> scan_chunk(const ChunkID chunk_id) override;
+  std::string description() const final;
 
  protected:
-  void _scan_generic_segment(const AbstractSegment& segment, const ChunkID chunk_id, RowIDPosList& matches) const;
+  void _scan_non_reference_segment(const AbstractSegment& segment, const ChunkID chunk_id, RowIDPosList& matches,
+                                   const std::shared_ptr<const AbstractPosList>& position_filter) final;
+
+  void _scan_generic_segment(const AbstractSegment& segment, const ChunkID chunk_id, RowIDPosList& matches,
+                             const std::shared_ptr<const AbstractPosList>& position_filter) const;
+
   void _scan_generic_sorted_segment(const AbstractSegment& segment, const ChunkID chunk_id, RowIDPosList& matches,
+                                    const std::shared_ptr<const AbstractPosList>& position_filter,
                                     const SortMode sorted_by) const;
 
-  // Optimized scan on ValueSegments
-  void _scan_value_segment(const BaseValueSegment& segment, const ChunkID chunk_id, RowIDPosList& matches);
-
   /**
-   * @defgroup Methods used for handling value segments
+   * @defgroup Method used for faster handling of value, dictionary, lz4, and frame_of_reference segments 
```

Comment:
`Method_s_`?
Unify naming (no snake case for FOR) and i'd say it's LZ4 (uppercase) because that's the official name.

Reply:
Is dash-cash correct for FOR?